### PR TITLE
Ft user booking accomodation #184638022

### PR DESCRIPTION
**Pull request by cyusakeny**
## What does this PR do?
 
User can book the accommodation he/she wants at a given time.

 ## Description of Task to be completed?

1. User can book a room
2. Room can't be booked twice in the same period.
 
 ##  How should this be manually tested?

Use the following route: Post http://localhost:8000/api/v1/booking/:roomId

 ##  Any background context you want to provide?

1. You have to be logged in to book accommodation.
2. You have to provide the date to come and the date to leave

 ## What are the relevant pivotal tracker/Trello stories?
Users should be able to book an accommodation facility.
[#184638022](https://github.com/atlp-rwanda/rca-barfoot-team3-bn/tree/ft-userBookingAccomodation-%23184638022)

 ## Screenshots
While inserting data:
image

On success:
image

Booking a room that was already booked:
image

### CODE REVIEW

**Comment by Irakoze-Hortance** [comment]
I think we need to have a route to get all rooms

File: src/modules/user/controller/index.js [hunk 1]
```diff
@@ -200,34 +199,20 @@ async function updateUserById(req, res) {
...
-    user.birthdate = birthdate;
-    user.username = username;
-    user.preferred_currency = preferred_currency;
-    user.preferred_language = preferred_language;
-    user.address = address;
-    user.role = role;
-    user.department = department;
-    user.lineManager = lineManager;
-
+    // eslint-disable-next-line no-restricted-syntax, guard-for-in
```
Comment (hirwablessing) [inline src/modules/user/controller/index.js]: @cyusakeny please fix this lint error instead of commenting it. We need to follow all the linting rules.

File: src/app.js [hunk 2]
```diff
@@ -10,8 +10,10 @@ require('./config/passport');
 require('dotenv').config();
 const { dbClient } = require('./database/index');
 const usersRouter = require('./modules/user/routes');
+
 const accommodationRouter = require('./modules/accommodations/routes');
 const swaggerConfig = require('../swagger.json');
+const bookingRoute = require('./modules/Booking/routes');
```
Comment (hirwablessing) [inline src/app.js]: since there will be multiple routes you might consider renaming the variable as `bookingRoutes` since this increases readability and show that there are multiple routes for booking there.

File: .env.example
Comment (hirwablessing) [inline .env.example:1]: We agreed on using yarn as our package manager so please stick with and don't mix it with npm. Delete this file and make sure that you install all dependencies with yarn.

File: .gitignore [hunk 3]
```diff
@@ -360,7 +360,9 @@ MigrationBackup/
...
 .env
-yarn.lock
+yarn
 
 .vscode
-views
\ No newline at end of file
+views
+yarn.lock
+package-lock.json
```
Comment (hirwablessing) [inline .gitignore]: You can remove this line once you've set your local project to use yarn

File: .env.example [hunk 4]
```diff
@@ -4,11 +4,10 @@ MESSAGE=
 
 # Database Configurations
 
-DB_NAME=
```
Comment (hirwablessing) [inline .env.example:7]: why did you remove this though?

**Review by hirwablessing — CHANGES_REQUESTED** [review]
@cyusakeny check the comments I left and let me know once you're done working on them.

**Review by DeliceLydia — APPROVED** [review]
